OptimizelySDKCore/OptimizelySDKCore/OPTLYDecisionService.m

```diff
@@ -132,11 +122,151 @@ - (OPTLYVariation *)getVariation:(NSString *)userId
     return bucketedVariation;
 }
 
+- (OPTLYFeatureDecision *)getVariationForFeature:(OPTLYFeatureFlag *)featureFlag
+                                    userId:(NSString *)userId
+                                attributes:(NSDictionary *)attributes {
+    
+    //Evaluate in this order:
+    
+    //1. Attempt to bucket user into experiment using feature flag.
+    // Check if the feature flag is under an experiment and the the user is bucketed into one of these experiments
+    OPTLYFeatureDecision *decision = [self getVariationForFeatureExperiment:featureFlag userId:userId attributes:attributes];
+    if (decision) {
+        return decision;
+    }
+    
+    //2. Attempt to bucket user into rollout using the feature flag.
+    // Check if the feature flag has rollout and the user is bucketed into one of it's rules
+    decision = [self getVariationForFeatureRollout:featureFlag userId:userId attributes:attributes];
+    if (decision) {
+        return decision;
+    }
+    [self.config.logger logMessage:[NSString stringWithFormat:OPTLYLoggerMessagesDecisionServiceFRUserNotBucketed, userId, featureFlag.Key]
+                         withLevel:OptimizelyLogLevelDebug];
+    
+    return nil;
+}
+
```

Comment:
OPTLYFeatureFlag AND getVariationForFeature:  NEED "groupId" 

OPTLYFeatureFlag is missing "groupId" and getVariationForFeature: needs to be coded
more like python-sdk's optimizely/decision_service.py get_variation_for_feature which
uses "groupId".  The "groupId" is not mentioned in our design doc, but is computed
client side.  I discussed the feature flag "groupId" with Ali Rizvi who supplied me with
this information.  The java-sdk, our other "reference" SDK, still has to catch up with
the python-sdk regarding this fine detail.  You will find "class FeatureFlag" in the
python-sdk's "entities.py" .  Use this as a guide for OPTLYFeatureFlag.m+.h which
currently doesn't expose "groupId".